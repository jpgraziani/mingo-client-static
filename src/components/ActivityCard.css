.activity-card {
  background-color: var(--lightBlue);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 375px;
  width: 340px;
  transition: all 0.3s ease-out;

}

.activity-card:hover {
  transform: translateY(-5px) scale(1) translateZ(0);
}

.title-card {
  display: flex;
  width: 320px;
  overflow: hidden;
  padding: 0 1.5rem;
  text-transform: capitalize;
}

.title-card a:link {
  color: var(--primary);
  text-decoration: none;
}

.preview-text {
  padding: 0 1.5rem;
  font-size:1.5rem;
  width:320px;
  
  /** Properties **/
  overflow:hidden;
  line-height: 2rem;
  max-height: 8rem;
  -webkit-box-orient: vertical;
  display: block;
  display: -webkit-box;
  overflow: hidden !important;
  text-overflow: ellipsis;
  -webkit-line-clamp: 2;
}
.activity-card h2 {
  color: var(--primary);
  margin: 0;
  margin-bottom: 1rem;
  font-size: 2.25rem;
  line-height: 2.25rem;
}

.activity-card p {
  font-size: 1.5rem;
  margin: 0;
  color: var(--primary);
}

.activity-card textarea::placeholder {
  color: rgba(250, 250, 250, .6);
}

.activity-card .hr {
  width: 90%;
  margin-bottom: 1rem;
  height: 3px;
  background-color: var(--primary);
}

.activity-card .lower-hr {
  height: 1.5px;
}
.click-card-container {
  display: flex;
}

.activity-card-btn {
  display: inline-block;
	margin-top: 1.5rem;
	margin-right: 1rem;
	padding: .9rem 1rem;
	text-decoration: none;
	font-size: 1rem;
	letter-spacing: 1px;
	font-weight: 600;
  width: 105px;
}

.input-btn {
  background: none;
  border: none;
  border: 2px solid var(--primary);
  color: var(--primary);
}

.activity-card-btn:link,
.activity-card-btn:visited {
	border: 2px solid var(--primary);
	color: var(--primary);
}

.activity-card-btn:hover{
	border: 2px solid var(--medBlue);
	background-color: var(--medBlue);
	color: var(--white);
}