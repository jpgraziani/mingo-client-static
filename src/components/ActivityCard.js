import React from 'react';
import { Link } from 'react-router-dom'
import ApiContext from '../ApiContext'
import config from '../config'

export default class ActivityCard extends React.Component {
  static defaultProps = {
    deleteActivity: () => {}
  }

  static contextType = ApiContext;

  handleDelete = e => {
    e.preventDefault()

    const activityId = this.props.id

    fetch(`${config.API_ENDPOINT}/activities/${activityId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      }
    })
    .then(res => {
      if (!res.ok) 
        return res.json().then(e => Promise.reject(e))
        return res.json()
    })
    .then(() => {
      this.context.deleteActivity(activityId)
      this.props.deleteActivity(activityId)
    })
    .catch(error => {
      console.error({ error })
    })
  }

  render() {
    const { name, id, date } = this.props
    return (
      <section>
        <div className='activity-card'>
          <Link to={`/activity/${id}`}>
            <h2>{name}</h2>
            <p>{date}</p>
          </Link>
          <button
            className='btns btn-full'
            type='button'
            onClick={this.handleDelete}
          >
            remove
          </button>
        </div>
      </section>
    );
  }
}