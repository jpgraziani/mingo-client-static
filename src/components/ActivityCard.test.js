import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ActivityCard from './ActivityCard'

describe(`ActivityCard component`, () => {
  it('renders a .ActivityCard by default', () => {
    const wrapper = shallow(<ActivityCard />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})