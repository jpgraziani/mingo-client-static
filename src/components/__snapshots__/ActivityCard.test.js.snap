// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityCard component renders a .ActivityCard by default 1`] = `
<section>
  <div
    className="activity-card card"
  >
    <div
      className="hr"
    />
    <div
      className="title-card"
    >
      <h2
        className="title-text"
      />
    </div>
    <div
      className="hr lower-hr"
    />
    <div
      className="preview-card hidden"
    >
      <p
        className="preview-text"
      />
    </div>
    <div
      className="click-card-container"
    >
      <div>
        <Link
          className="activity-card-btn act-link"
          to="/activity/undefined"
        >
          read me
        </Link>
      </div>
      <div>
        <input
          className="activity-card-btn input-btn"
          onClick={[Function]}
          type="button"
          value="delete"
        />
      </div>
    </div>
  </div>
</section>
`;
